admin-portal/src/app/_components/EventCard.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useState, useEffect } from 'react';
+import axios, { AxiosResponse } from "axios";
+import EditCard from "./EditCard";
+
+interface EventCardProps {
+    id: string;
+    removeFromList: (id: string) => void;
+}
+
+// EventData interface
+interface ActivityContent {
+    [key: string]: unknown;
+}
+
+interface Activity {
+    type: string;
+    content: ActivityContent;
+    timeStart: Date;
+    timeEnd: Date;
+    active: boolean;
+}
+
+interface EventData {
+    _id: string;
+    name: string;
+    date: Date;
+    description: string;
+    location: {
+        type: string;
+        coordinates: number[];
+    };
+    organizerId: string;
+    tags: string[];
+    registeredUsers: string[];
+    activities: Activity[];
+}
```

Comment:
For common interfaces used in different files (i.e. for this one, it is EventCard and OrganizationProfile), move these into their own dedicated file. Create a new folder under app called _interfaces and a file named `EventInterfaces.tsx` which exports each of these shared interfaces. 

This reduces code redundancy overall! But good job overall defining these and especially using `unknown` which is generally safer than `any` as it still enforces type-checks later on.